packages/cozy-client/src/__tests__/store.spec.js

```diff
@@ -178,6 +179,57 @@ describe('Store', () => {
     })
   })
 
+  describe('getCollectionFromState', () => {
+    it('should return null when the store is empty', () => {
+      const spy = jest.spyOn(global.console, 'warn').mockReturnValue(jest.fn())
+      expect(getCollectionFromState(store.getState(), 'io.cozy.todos')).toBe(
+        null
+      )
+      spy.mockRestore()
+    })
+
+    it('should return document list for given doctype', () => {
+      store.dispatch(
+        receiveQueryResult('allTodos', {
+          data: [TODO_1, TODO_2],
+          meta: { count: 2 },
+          skip: 0,
+          next: false
+        })
+      )
+      expect(getCollectionFromState(store.getState(), 'io.cozy.todos')).toEqual(
+        [TODO_1, TODO_2]
+      )
+    })
+
+    it('should return a list containing a newly created doc received in a mutation result', () => {
+      store.dispatch(
+        receiveMutationResult('foo', {
+          data: [TODO_1]
+        })
+      )
+      expect(getCollectionFromState(store.getState(), 'io.cozy.todos')).toEqual(
+        [TODO_1]
+      )
+    })
+
+    it('should return an updated rev of a doc received in a mutation result', () => {
+      store.dispatch(
+        receiveMutationResult('foo', {
+          data: [TODO_1]
+        })
+      )
```

Comment:
I am not sure I would test this logic. It is the role of receiveMutationResult to correctly update the store. Here we are testing receiveMutationResult instead of getCollectionFromState.

The 2 first tests seem sufficient.

Reply:
I agree, I adapted the tests for `getDocumentFromState` but they are not all pertinent for `getCollectionFromSate`.